packages/compass-workspace/src/components/workspace-tabs.tsx

```diff
@@ -0,0 +1,347 @@
+        evt.stopPropagation();
+        nextTabbable = selectedTabIndex - 1;
+      }
+
+      if (evt.key === 'ArrowRight') {
+        evt.stopPropagation();
+        nextTabbable = selectedTabIndex + 1;
+      }
+
+      if (evt.key === 'Home') {
+        evt.preventDefault();
+        evt.stopPropagation();
+        // Select the first tab.
+        nextTabbable = 0;
+      }
+
+      if (evt.key === 'End') {
+        evt.preventDefault();
+        evt.stopPropagation();
+        // Select the last tab.
+        nextTabbable = tabsCount - 1;
+      }
+
+      if (
+        tabsCount > 0 &&
+        nextTabbable !== selectedTabIndex &&
+        nextTabbable >= 0 &&
+        nextTabbable < tabsCount
+      ) {
+        onTabSelected(nextTabbable);
+      }
+    },
+    [selectedTabIndex, tabsCount, onTabSelected]
+  );
+
+  return [{ onKeyDown }];
+}
+
+type SortableItemProps = {
+  tab: TabProps;
+  tabIndex: number;
+  isTabListFocused: boolean;
+  onSelect: (tabIndex: number) => void;
+  onClose: (tabIndex: number) => void;
+};
+
+const SortableItem = SortableElement(
+  ({
+    tab: {
+      activeSubTabName,
+      isActive,
+      id: tabId,
+      namespace,
+      isTimeSeries,
+      isReadonly,
+    },
+    tabIndex,
+    isTabListFocused,
+    onSelect,
+    onClose,
+  }: SortableItemProps) => {
+    const onTabSelected = useCallback(() => {
+      onSelect(tabIndex);
+    }, [onSelect, tabIndex]);
+
+    const onTabClosed = useCallback(() => {
+      onClose(tabIndex);
+    }, [onClose, tabIndex]);
+
+    const tabType = useMemo(
+      () => getTabType(isTimeSeries, isReadonly),
+      [isTimeSeries, isReadonly]
+    );
+
+    return (
+      <Tab
+        activeSubTabName={activeSubTabName}
+        isSelected={isActive}
+        isFocused={isTabListFocused && isActive}
+        tabId={tabId}
+        type={tabType}
+        namespace={namespace}
+        onSelect={onTabSelected}
+        onClose={onTabClosed}
+        isTabListFocused={isTabListFocused}
+      />
+    );
+  }
+);
+
+type SortableListProps = {
+  tabs: TabProps[];
+  isTabListFocused: boolean;
+  onSelect: (tabIndex: number) => void;
+  onClose: (tabIndex: number) => void;
+};
+
+const SortableList = SortableContainer(
+  ({ tabs, isTabListFocused, onSelect, onClose }: SortableListProps) => (
+    <div className={sortableItemContainerStyles}>
+      {tabs.map((tab: TabProps, index: number) => (
+        <SortableItem
+          key={`tab-${index}-${tab.namespace}`}
+          // `index` is used internally by the SortableContainer hoc,
+          // so we pass our own `tabIndex`.
+          index={index}
+          tabIndex={index}
+          tab={tab}
+          isTabListFocused={isTabListFocused}
+          onSelect={onSelect}
+          onClose={onClose}
+        />
+      ))}
+    </div>
+  )
+);
+
+type TabProps = {
+  namespace: string;
+  id: string;
+  activeSubTabName: string;
+  isActive: boolean;
+  isTimeSeries: boolean;
+  isReadonly: boolean;
```

Comment:
Does it makes sense to have this as part of the component interface? I know that it was like that before, but aren't we only interested in collection type here (like view or timeseries) similar to how it is done in the sidebar navigation

Reply:
To go one step forward, is there a way to completely split the code that "knows" about the domain of what the tab content is doing from the one that renders / moves and locates the tabs?

I would imagine something like this (ideally and very pseudo-codeish):

```
<WorkspaceTabs tabs={[
    {
       id: 'my-tab-123',
       header: <CollectionTabHeader ns="my-db.my-coll" type="timeseries"> 
       content: <CollectionTabContent selectedSubTab={...}>
    }
]}>
```

Reply:
Yeah, something like this would be even better, I agree

Reply:
Good thinking, I've updated to make most of the component more generic and accept an icon.
I think it might make sense to avoid moving the content rendering into this `compass-workspace` in this pr, I'll update it so that we're passing a render function from `compass-collection` as `compass-collection` has a good amount of logic tied around how the tabs are generated and even some ipc listeners for interacting with the tabs.

Reply:
Sounds good, created a ticket in the epic to move more of the workspace into the package and make the boundary between `Collection` view and `Workspace` view a bit more distinct: https://jira.mongodb.org/browse/COMPASS-5605